pkg/controllers/status/cluster_status_controller.go

```diff
@@ -197,21 +195,29 @@ func (c *ClusterStatusController) syncClusterStatus(cluster *clusterv1alpha1.Clu
 		} else if err != nil {
 			klog.Warningf("Maybe get partial(%d) APIs installed in Cluster %s. Error: %v.", len(apiEnables), cluster.GetName(), err)
 		}
+		currentClusterStatus.APIEnablements = apiEnables
 
-		nodes, err := listNodes(clusterInformerManager)
-		if err != nil {
-			klog.Errorf("Failed to list nodes for Cluster %s. Error: %v.", cluster.GetName(), err)
-		}
+		if c.EnableClusterResourceModeling {
+			// get or create informer for pods and nodes in member cluster
+			clusterInformerManager, err := c.buildInformerForCluster(clusterClient)
```

Comment:
The `buildInformerForCluster` now helps to initialize the informer for clusters, the informer will be used by other controllers. 
We must de-coupling it before moving this forward.
/hold

Reply:
+1
The current control coupling is a little severe.

Reply:
Maybe we can build the `generic informer` as usual but not build the `typed informer` in `buildInformerForCluster`?

Reply:
That's definitely a workable solution. 

But the annoying thing is it's a bad program paradigm. Too much coupling leads to hard to maintain.
Now, the `multi-cluster-informer-manager` is initialized by `cluster-status-controller` but used by `execution-controller` , and `work-status-controller`. :(

I'm trying to decouple them.